scripts/linters/pylint_extensions.py

```diff
@@ -2919,6 +2919,52 @@ def visit_importfrom(self, node: astroid.nodes.ImportFrom) -> None:
                 self.add_message('disallowed-text-import', node=node)
 
 
+# TODO(#16567): Here we use MyPy ignore because of the incomplete typing of
+# pylint library and absences of stubs in pylint, forces MyPy to
+# assume that BaseChecker class has attributes of type Any.
+# Thus to avoid MyPy's error
+# (Class cannot subclass 'BaseChecker' (has type 'Any')),
+# we added an ignore here.
+class PreventStringConcatenationChecker(checkers.BaseChecker): # type: ignore[misc]
+    """Checks for string concactenation and encourage string interpolation."""
+
+    __implements__ = interfaces.IAstroidChecker
+
+    name = 'use-string-interpolation'
+    priority = -1
+    msgs = {
+        'C0042': (
+            'Please use string interpolation over string concatenation',
+            'use-string-interpolation',
+            'Used when string concatenation is detected.',
+        ),
+    }
+
+    def visit_binop(self, node: astroid.BinOp) -> None:
+        """Visits a binary operation node in the AST
+        then checks if it's a string concatenation operation.
+
+        Args: node: astroid.BinOp The binary operation node to check.
+        """
+
+        if isinstance(node, astroid.BinOp) and node.op == '+':
```

Comment:
```suggestion
        if !isinstance(node, astroid.BinOp) or node.op != '+':
            return
```
Optional*

Use it as a guard clause and early return. (Also, helps avoid the big indented block below)

Reply:
Done.

Reply:
Not done? (I don't see the changes in the code.)

Reply:
After I made the change, I was getting code coverage error. So, I kept prior code